WORKSPACE

```diff
@@ -5,21 +5,35 @@ workspace(
     managed_directories = {"@npm": ["ui/node_modules"]},
 )
 
-load("//bazel:deps.bzl", "enkit_deps")
+load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
 
+load("//bazel:deps.bzl", "enkit_deps")
 enkit_deps()
 
-load("//bazel:init.bzl", "enkit_init")
+# packages must be loaded in this staggered manner to fix the dependency issue
+# in order for protobuf and grpc to work hence the removal of enkit_init()
```

Comment:
can't the steps here go into enkit_init()? Eg, you change the implementation of enkit_init() to just have the steps here in this order?

The underlying problem is that the enkit repository is used from within other repositories. The initializations below need to be done in other repositories as well. 
The typical way to solve this problem is to have those initializations in a macro, that then any WORKSPACE can use.

So... if we get rid of enkit_init here, everyone else using this repository will need to cut and paste the lines. Which will make updates messy / hard to do, as now everyone will need to also update the copy of their rules.

Reply:
If you don't load and initialize the bazel packages in this exact order, you get this error 
```
ERROR: /home/bbhuynh/.cache/bazel/_bazel_bbhuynh/ccbc8eec90d71932e4b174477612b27d/external/io_bazel_rules_nogo/BUILD.bazel:3:6: in alias rule @io_bazel_rules_nogo//:nogo: cycle in dependency graph:
    //kbuild/kapt:go_default_library
    //lib/khttp/protocol:go_default_library
    //lib/khttp/kclient:go_default_library
    @io_bazel_rules_go//:go_context_data
.-> @io_bazel_rules_nogo//:nogo
|   @io_bazel_rules_go//:default_nogo
|   @io_bazel_rules_go//go/tools/builders:nogo_srcs
|   @org_golang_x_tools//go/analysis/internal/facts:go_tool_library
|   @org_golang_x_tools//go/analysis:go_tool_library
|   @org_golang_x_tools//internal/analysisinternal:go_tool_library
|   @org_golang_x_tools//go/ast/astutil:go_tool_library
|   @org_golang_x_tools//internal/typeparams:typeparams
|   @io_bazel_rules_go//:go_context_data
`-- @io_bazel_rules_nogo//:nogo
```
which is being fixed by the bazel developers and will probably be in the next release according to the activity on 
https://github.com/bazelbuild/bazel/issues/11291
https://github.com/bazelbuild/bazel/pull/13372
so let's just get this merged, then we can revisit this later once the bazel fix is in the next bazel release.

Reply:
I'm not arguing about changing the order, I'm arguing about moving this in the `enkit_init()` function in the .bzl file.
The order would be the same?

Reply:
Moving it in `enkit_init` in the same order would also allow you to avoid a few changes in https://github.com/enfabrica/internal/pull/996, eg, would save you from duplicating the protobuf configuration.